File: Arithmetic Slices-Brute Force.cpp
//Time Complexity-O(n^2)
//Space Complexity-O(1)
//Did the code execute on Leetcode? Yes

class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& A) {
        if(A.size()==0 || A.size()<3)
        {
            return 0;
        }
        int count=0;
        for(int i=0;i<A.size()-2;i++)
        {
            int a=A[i+1]-A[i];
            for(int j=i+1;j<A.size()-1;j++)
            {
                if(A[j+1]-A[j]==a)
                {
                    count++;
                }
                else
                {
                    break;
                }
            }
                
        }
        return count;
    }
};

File: arithmetic_slices_bruteforce.py
# Time Complexity : O(N*N) where N is length of the nums array
# Space Complexity : O(1) 
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        count = 0
        for i in range(len(nums)-2):
            diff = nums[i+1] - nums[i]
            for j in range(i+2, len(nums)):
                if nums[j] - nums[j-1] == diff:
                    count += 1
                else:
                    break      
        return count